crypto-ballerina/src/crypto/tests/unit-tests/rsa-ecb-encrypt-decrypt-test.bal

```diff
@@ -0,0 +1,143 @@
+}
+
+@test:Config {}
+function testEncryptAndDecryptWithRsaEcbOaepWithSha384andMgf1() {
+    byte[] message = "Ballerina crypto test           ".toBytes();
+    KeyStore keyStore = {
+        path: "src/crypto/tests/resources/datafiles/testKeystore.p12",
+        password: "ballerina"
+    };
+    PublicKey puk = checkpanic decodePublicKey(keyStore, "ballerina");
+    PrivateKey prk = checkpanic decodePrivateKey(keyStore, "ballerina", "ballerina");
+    byte[] cipherText = checkpanic encryptRsaEcb(message, puk, "OAEPwithSHA384andMGF1");
+    byte[] plainText = checkpanic decryptRsaEcb(cipherText, prk, "OAEPwithSHA384andMGF1");
+    test:assertEquals(plainText.toBase16(), message.toBase16(),
+        msg = "Error while Encrypt/Decrypt with RSA ECB OAEPwithSHA384andMGF1.");
+}
+
+@test:Config {}
+function testEncryptAndDecryptWithRsaEcbOaepWithSha512andMgf1() {
+    byte[] message = "Ballerina crypto test           ".toBytes();
+    KeyStore keyStore = {
+        path: "src/crypto/tests/resources/datafiles/testKeystore.p12",
+        password: "ballerina"
+    };
+    PublicKey puk = checkpanic decodePublicKey(keyStore, "ballerina");
+    PrivateKey prk = checkpanic decodePrivateKey(keyStore, "ballerina", "ballerina");
+    byte[] cipherText = checkpanic encryptRsaEcb(message, puk, "OAEPwithSHA512andMGF1");
+    byte[] plainText = checkpanic decryptRsaEcb(cipherText, prk, "OAEPwithSHA512andMGF1");
+    test:assertEquals(plainText.toBase16(), message.toBase16(),
+        msg = "Error while Encrypt/Decrypt with RSA ECB OAEPwithSHA512andMGF1.");
+}
+
+@test:Config {}
+function testEncryptWithPrivateKeyAndDecryptWithPublicKeyUsingRsaEcbPkcs1() {
+    byte[] message = "Ballerina crypto test           ".toBytes();
+    KeyStore keyStore = {
+        path: "src/crypto/tests/resources/datafiles/testKeystore.p12",
+        password: "ballerina"
+    };
+    PublicKey puk = checkpanic decodePublicKey(keyStore, "ballerina");
+    PrivateKey prk = checkpanic decodePrivateKey(keyStore, "ballerina", "ballerina");
+    byte[] cipherText = checkpanic encryptRsaEcb(message, prk, "PKCS1");
+    byte[] plainText = checkpanic decryptRsaEcb(cipherText, puk, "PKCS1");
+    test:assertEquals(plainText.toBase16(), message.toBase16(),
+        msg = "Error while Encrypt/Decrypt with RSA ECB PKCS1.");
+}
+
+@test:Config {}
+function testEncryptAndDecryptWithRsaEcbPkcs1WithAnInvalidKey() {
+    byte[] message = "Ballerina crypto test           ".toBytes();
+    PrivateKey invalidPrk = {algorithm:"RSA"};
+    byte[]|error result = encryptRsaEcb(message, invalidPrk, "PKCS1");
+    if (result is error) {
+        test:assertEquals(extractErrorMessage(result), "Uninitialized private/public key",
+            msg = "Incorrect error for for invalid key while Encryption with RSA ECB.");
+    } else {
+        test:assertFail(msg = "No error for invalid key Encryption with RSA ECB PKCS1.");
+    }
+}
+
```

Comment:
The test case for encrypt and decrypt with RSA ECB PKCS1 with invalid padding is missing here. Shall we add that also?